### PR TITLE
Fix various memory leaks across changing tabs and opening new sources

### PR DESCRIPTION
**Pull request by snosenzo**
**User-Facing Changes**

**Description**

Fixes: #4824

### CODE REVIEW

File: packages/studio-base/src/panels/ThreeDeeRender/Renderer.ts [hunk 1]
```diff
@@ -479,31 +479,41 @@ export class Renderer extends EventEmitter<RendererEvents> {
...
   public dispose(): void {
     log.warn(`Disposing renderer`);
+    this._devicePixelRatioMediaQuery?.removeEventListener("change", this._onDevicePixelRatioChange);
     this.removeAllListeners();
 
     this.settings.off("update");
+    this.settings.removeAllListeners();
     this.input.off("resize", this.resizeHandler);
     this.input.off("click", this.clickHandler);
+    this.input.removeAllListeners();
```
Comment (jhurliman) [inline packages/studio-base/src/panels/ThreeDeeRender/Renderer.ts:516]: Can the `this.settings.off("update");` and `this.input.off(...)` calls be removed?

File: packages/studio-base/src/panels/ThreeDeeRender/ThreeDeeRender.tsx [hunk 2]
```diff
@@ -558,6 +556,11 @@ export function ThreeDeeRender({ context }: { context: PanelExtensionContext }):
     context.watch("parameters");
     context.watch("variables");
     context.watch("topics");
+    return () => {
+      // release panel extension adapter from old panel
+      context.onRender = undefined;
+      context.unsubscribeAll();
```
Comment (jhurliman) [inline packages/studio-base/src/panels/ThreeDeeRender/ThreeDeeRender.tsx:562]: Does every extension panel need to do this?

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/Axis.ts [hunk 3]
```diff
@@ -60,8 +71,10 @@ export class Axis extends THREE.Object3D {
   }
 
   public dispose(): void {
+    this.shaftMesh.geometry.dispose();
```
Comment (jhurliman) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/Axis.ts:74]: Why create a copy of the geometry just to dispose it?

File: packages/studio-base/src/panels/ThreeDeeRender/renderables/Axis.ts [hunk 4]
```diff
@@ -103,6 +116,9 @@ export class Axis extends THREE.Object3D {
 
   private static ShaftGeometry(lod: DetailLevel): THREE.CylinderGeometry {
     if (!Axis.shaftGeometry || lod !== Axis.shaftLod) {
+      if (Axis.shaftGeometry) {
+        Axis.shaftGeometry.dispose();
+      }
```
Comment (jhurliman) [inline packages/studio-base/src/panels/ThreeDeeRender/renderables/Axis.ts:121]:
```suggestion
      Axis.shaftGeometry?.dispose();
```

File: packages/studio-base/src/components/PanelExtensionAdapter/PanelExtensionAdapter.tsx [hunk 5]
```diff
@@ -490,7 +491,9 @@ function PanelExtensionAdapter(props: PanelExtensionAdapterProps): JSX.Element {
     });
 
     return () => {
+      ReactDOM.unmountComponentAtNode(panelElement);
```
Comment (defunctzombie) [inline packages/studio-base/src/components/PanelExtensionAdapter/PanelExtensionAdapter.tsx:494]: Note that extension panels do not have to use react so this is the responsibility of the panel author not our extension interfaces to unmount their react node.

**Comment by snosenzo** [comment]
closing as a lot of this was pulled into separate pull requests all around solving #4824

**Pull request closed without merging**